console/console-init/ui/src/components/AdapterList/AdapterList.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React from "react";
+import { Title, Grid, GridItem } from "@patternfly/react-core";
+import { InputText, StatusLabelWithIcon } from "components";
+import { Protocols } from "constant";
+
+export interface IAdapterConfig {
+  url?: string;
+  host?: string;
+  port?: number;
+  tlsEnabled?: boolean;
+}
+
+export interface IAdapter {
+  type:
+    | Protocols.HTTP
+    | Protocols.HTTPS
+    | Protocols.MQTT
+    | Protocols.AMQP
+    | Protocols.AMQPS
+    | Protocols.COAP;
+  value: IAdapterConfig;
+}
+
+export interface IAdapterListProps {
+  id: string;
+  adapters: IAdapter[];
+}
+
+export const AdapterList: React.FC<IAdapterListProps> = ({ id, adapters }) => {
+  return (
+    <div id={id}>
+      {adapters &&
+        adapters.map((adapter: IAdapter, index: number) => (
+          <Adapter adapter={adapter} key={`adapter-${adapter.type}`} />
+        ))}
+    </div>
+  );
+};
+
+const Adapter: React.FC<{ adapter: IAdapter }> = ({ adapter }) => {
+  const { type, value } = adapter || {};
+  const { tlsEnabled, host, port, url } = value || {};
+
+  return (
+    <>
+      <Title size="xl" headingLevel="h2">
+        {type.toUpperCase() + " Adapter"}
```

Comment:
null | undefined check for type is missing

Reply:
yes, I do. This change will come in next PR.